homeassistant/components/zeroconf/__init__.py

```diff
@@ -458,10 +459,17 @@ async def _process_service_update(
 
             matcher_domain = matcher["domain"]
             assert isinstance(matcher_domain, str)
+            context = {
```

Comment:
Unrelated but hoping at some point we can make `_process_service_update` a callback function since it creates 300-600 tasks/min

Reply:
wow

Reply:
It's async because of `await async_service_info.async_request(zeroconf, 3000)` Do we need to call that? If we just got notified of an update about it, can't we assume we're up to date?

Reply:
99% of the time we can actually use `load_from_cache` instead and only spawn a task if we don't have it

`zeroconf/_services/info.py:    def load_from_cache(self, zc: 'Zeroconf') -> bool:`

The other reason is the async_get_integration call below it